executor/union_scan.go

```diff
@@ -267,6 +269,24 @@ func (us *UnionScanExec) compare(a, b []types.Datum) (int, error) {
 	return cmp, nil
 }
 
+// rowWithColsInTxn gets the row from the transaction buffer.
+func (us *UnionScanExec) rowWithColsInTxn(t table.Table, h int64, cols []*table.Column) ([]types.Datum, error) {
+	key := t.RecordKey(h)
+	txn, err := us.ctx.Txn(true)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	value, err := txn.GetMemBuffer().Get(key)
```

Comment:
Can we pass a parameter into `RowWithCols` to differentiate the behavior here and avoid duplicate code? or can we set `PresumeKeyNotExists` and `PresumeKeyNotExistsError` options accordingly to get the behavior we want, i.e, just check transaction buffer?

Reply:
Setting `PresumeKeyNotExists` is a bit too far. It will write the lazy pair condition again, which should have been written once in the previous insert. Adding options to `RowWithCols`, I think it's not worth changing the interface. In fact, the repetition code is not too much here, and many lines are occupied by error handling.